src/main/java/teammates/common/util/ActivityLogGenerator.java

```diff
@@ -0,0 +1,336 @@
+package teammates.common.util;
+
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.TimeZone;
+import java.util.regex.Pattern;
+
+import javax.servlet.http.HttpServletRequest;
+
+import com.google.appengine.api.log.AppLogLine;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
```

Comment:
This dependency is used when exception/system error is caught. In that situation, if we want to know who is the current login user causing the exception/system error, it is necessary to introduce the `UserService` here.